src/main/java/Duke.java

```diff
@@ -34,17 +111,51 @@ private static void listTasks() {
 
         for (int i = 0; i < taskManager.getTotalTasks(); i++) {
             Task task = taskManager.getTask(i);
-            taskListMessage[i + 1] = (i + 1) + ".[" + task.getStatusIcon() + "] " + task.getDescription();
+            taskListMessage[i + 1] = (i + 1) + ". " + task.toString();
         }
 
         blockPrint(taskListMessage);
     }
 
     private static void markTaskAsDone(int taskIndex) {
+        if (taskManager.getTotalTasks() == 0) {
+            blockPrint(new String[]{"The list is currently empty. Add a task first."});
+            return;
+        } else if (taskIndex < 0 || taskIndex >= taskManager.getTotalTasks()) {
+            blockPrint(new String[]{"Invalid task index."});
+            return;
+        }
+
         taskManager.markTaskAsDone(taskIndex);
         Task completedTask = taskManager.getTask(taskIndex);
         blockPrint(new String[]{"Affirmative. I will mark this task as done:",
-                "[" + completedTask.getStatusIcon() + "] " + completedTask.getDescription()});
+                "[" + completedTask.getType() + "][" + completedTask.getStatusIcon() + "] "
+                        + completedTask.getDescription()});
+    }
+
+    private static void parseCommand(String[] splitUserInput) {
+        String userCommand = splitUserInput[0];
+        switch (userCommand) {
+        case COMMAND_LIST:
+            listTasks();
+            break;
+        case COMMAND_DONE:
+            int taskIndex = Integer.parseInt(splitUserInput[1]) - 1;
+            markTaskAsDone(taskIndex);
+            break;
+        case COMMAND_TODO:
+            addTodo(splitUserInput);
+            break;
+        case COMMAND_DEADLINE:
+            addDeadline(splitUserInput);
+            break;
+        case COMMAND_EVENT:
+            addEvent(splitUserInput);
+            break;
+        default:
+            printHelp();
+            break;
+        }
     }
```

Comment:
I like how you used case instead of if-else statements, then you will avoid the deep nesting :)